coalib/results/result_actions/IgnoreResultAction.py

```diff
@@ -57,11 +58,21 @@ def apply(self, result, original_file_dict, file_diff_dict, language: str,
         filename = source_range.file
 
         ignore_diff = Diff(original_file_dict[filename])
-        ignore_diff.change_line(
-            source_range.start.line,
-            original_file_dict[filename][source_range.start.line-1],
-            original_file_dict[filename][source_range.start.line-1].rstrip() +
-            '  ' + ignore_comment)
+
+        original_line = str(original_file_dict[filename][
+                        source_range.start.line-1].rstrip())
+
+        # Add the new ignore using 'and' with the previous ignore.
+        if re.search(r'Ignore.+Bear', original_line):
+            changed_line = re.sub(r'(.+Ignore\s)(.+)', r'\1' +
+                                  result.origin + r' and \2', original_line)
+
+        # Add a new ignore comment if no previous ignore exists.
+        else:
+            changed_line = original_line + '  ' + ignore_comment
+
+        ignore_diff.change_line(source_range.start.line, original_line,
+                                changed_line)
```

Comment:
Shouldn't you maybe modify `self.get_ignore_comment` instead?

Reply:
@Makman2 I have to just modify the previous comment. There is no need of getting a comment from the ``self.get_ignore_comment``. I only need the ``origin``. So, felt it best to modify ``self.apply``.

Reply:
I think modifying `get_ignore_comment` is the right thing to do. This function will return the ignore comment itself, and dependent on whether there were existing ignore comments, you assemble the line differently and generate the ignore comment differently (using `and` as a conjunction).

Reply:
Especially you use `get_ignore_comment` only in the case that there were no previous ignore comments, so it's currently a corner-case function :) And this shouldn't be the case ;)